--- dwm.c.orig	2011-12-19 13:02:46.000000000 -0200
+++ dwm.c	2013-04-02 03:05:51.406811923 -0300
@@ -717,7 +717,6 @@
 
 void
 drawbar(Monitor *m) {
-	int x;
 	unsigned int i, occ = 0, urg = 0;
 	unsigned long *col;
 	Client *c;
@@ -727,40 +726,39 @@
 		if(c->isurgent)
 			urg |= c->tags;
 	}
+
+	int lw = dc.w = m->ww;
+	if(m == selmon) { /* status is only drawn on selected monitor */
+		dc.w = TEXTW(stext);
+		lw = dc.x = (m->ww - dc.w) / 2;
+		drawtext(stext, dc.norm, False);
+	}
+
+	dc.x += dc.w;
+	dc.w = m->ww - dc.x;
+	drawtext(NULL, dc.norm, False);
+
 	dc.x = 0;
+	dc.w = TEXTW(m->ltsymbol);
+	drawtext(m->ltsymbol, dc.norm, False);
+	dc.x += dc.w;
 	for(i = 0; i < LENGTH(tags); i++) {
 		dc.w = TEXTW(tags[i]);
 		col = m->tagset[m->seltags] & 1 << i ? dc.sel : dc.norm;
 		drawtext(tags[i], col, urg & 1 << i);
 		drawsquare(m == selmon && selmon->sel && selmon->sel->tags & 1 << i,
 		           occ & 1 << i, urg & 1 << i, col);
 		dc.x += dc.w;
 	}
-	dc.w = blw = TEXTW(m->ltsymbol);
-	drawtext(m->ltsymbol, dc.norm, False);
-	dc.x += dc.w;
-	x = dc.x;
-	if(m == selmon) { /* status is only drawn on selected monitor */
-		dc.w = TEXTW(stext);
-		dc.x = m->ww - dc.w;
-		if(dc.x < x) {
-			dc.x = x;
-			dc.w = m->ww - x;
-		}
-		drawtext(stext, dc.norm, False);
+	dc.w = lw - dc.x;
+	if(m->sel) {
+		col = m == selmon ? dc.sel : dc.norm;
+		drawtext(m->sel->name, dc.norm, False);
+		drawsquare(m->sel->isfixed, m->sel->isfloating, False, col);
 	}
 	else
-		dc.x = m->ww;
-	if((dc.w = dc.x - x) > bh) {
-		dc.x = x;
-		if(m->sel) {
-			col = m == selmon ? dc.sel : dc.norm;
-			drawtext(m->sel->name, col, False);
-			drawsquare(m->sel->isfixed, m->sel->isfloating, False, col);
-		}
-		else
-			drawtext(NULL, dc.norm, False);
-	}
+		drawtext(NULL, dc.norm, False);
+	
 	XCopyArea(dpy, dc.drawable, m->barwin, dc.gc, 0, 0, m->ww, bh, 0, 0);
 	XSync(dpy, False);
 }
