--- config.def.h.orig	2011-12-19 13:02:46.000000000 -0200
+++ config.def.h	2013-03-31 21:55:45.420340193 -0300
@@ -12,6 +12,8 @@
 static const unsigned int snap      = 32;       /* snap pixel */
 static const Bool showbar           = True;     /* False means no bar */
 static const Bool topbar            = True;     /* False means bottom bar */
+static const unsigned int hgap      = 16;       /* Gap between horizontally adjacent windows */
+static const unsigned int vgap      = 16-16/3;  /* Gap between vertically adjacent windows */
 
 /* tagging */
 static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
--- dwm.c.orig	2013-03-31 20:50:30.495486487 -0300
+++ dwm.c	2013-03-31 21:44:15.148071200 -0300
@@ -1902,16 +1894,16 @@
 		mw = m->nmaster ? m->ww * m->mfact : 0;
 	else
 		mw = m->ww;
-	for(i = my = ty = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
+	for(i = my = ty = 0, c = nexttiled(m->clients); c; c = nexttiled(c->next), i++)
 		if(i < m->nmaster) {
 			h = (m->wh - my) / (MIN(n, m->nmaster) - i);
 			resize(c, m->wx, m->wy + my, mw - (2*c->bw), h - (2*c->bw), False);
-			my += HEIGHT(c);
+			my += HEIGHT(c) + vgap;
 		}
 		else {
 			h = (m->wh - ty) / (n - i);
-			resize(c, m->wx + mw, m->wy + ty, m->ww - mw - (2*c->bw), h - (2*c->bw), False);
-			ty += HEIGHT(c);
+			resize(c, m->wx + mw + hgap, m->wy + ty, m->ww - mw - (2*c->bw) - hgap, h - (2*c->bw), False);
+			ty += HEIGHT(c) + vgap;
 		}
 }
 
